packages/core/src/document.ts

```diff
@@ -113,49 +114,122 @@ class Document extends EventEmitter {
+    // TODO: Assert that doc contains only the genesis record
+    const id = doc.id
 
+    // Update document state to cached state if any
     const isStored = await pinStore.stateStore.exists(id)
     if (isStored) {
       doc._doctype.state = await pinStore.stateStore.load(id)
     }
 
-    if (doc._doctype.state == null) {
-      // apply genesis record if there's no state preserved
-      doc._doctype.state = await doc._doctypeHandler.applyRecord(record, doc._genesisCid, context)
+    await doc._register(opts)
+    return doc
+  }
+
+  /**
+   * Takes a document containing only the genesis record and syncs the document to the state as
+   * of the specific version requested.  Intentionally does not register the document so that it
+   * does not get notifications about newer versions, since we want a specific version here.
+   * @param doc - Document containing only the genesis record
+   * @param tip - CID of the Tip record that we want to load the document at
+   * @param dispatcher
+   * @param mustBeVersion - only anchor records count as valid document versions.  If 'mustBeVersion'
+   *   is false and 'tip' refers to a signed record, throws an error
+   */
+  static async _syncDocumentToTip<T extends Doctype> (
+      doc: Document,
+      tip: CID,
+      dispatcher: Dispatcher,
+      mustBeVersion): Promise<Document> {
+    // TODO: assert that doc.id.version == version
+    // TODO: Assert that doc contains only the genesis record
+
+    if (tip.equals(doc.id.cid)) {
+      // The version is the same as the genesis record CID, so nothing more to do after loading
+      // the genesis version of the document.
+      doc._doctype = DoctypeUtils.makeReadOnly<T>(doc.doctype as T)
+      return doc
     }
 
-    if (validate) {
-      const schema = await Document.loadSchema(context, doc._doctype)
-      if (schema) {
-        Utils.validate(doc._doctype.content, schema)
-      }
+    // Load the requested version record
+    const versionRecord = await dispatcher.retrieveRecord(tip)
+    if (versionRecord == null) {
+      throw new Error(`No record found for CID ${tip.toString()}`)
     }
 
-    await doc._register(opts)
+    // check if it's not an anchor record
+    if (mustBeVersion && versionRecord.proof == null) {
+      throw new Error(`Log record CID ${tip.toString()} does not refer to a valid version, which must correspond to an anchor record`)
+    }
+
+    await doc._handleTip(tip) // sync version
+
+    doc._doctype = DoctypeUtils.makeReadOnly<T>(doc.doctype as T)
     return doc
   }
 
   /**
-   * Validate Doctype against schema
+   * Loads the genesis record and builds a Document object off it, but does not register for updates
+   * or apply any additional records past the genesis record.
+   * @param id - Document id
+   * @param handler
+   * @param dispatcher
+   * @param pinStore
+   * @param context
+   * @param validate
+   * @private
    */
-  async validate(): Promise<void> {
```

Comment:
This function was totally unused and was already a source of bugs via confusion between `this.validate` (the function) and `this._validate` (the boolean member variable).  Given validation should be happening as part of document loading anyway, I just went ahead and removed this function entirely